Dockerfile

```diff
@@ -1,9 +1,11 @@
 FROM node:6.11-alpine
 
-RUN mkdir /app
-ADD package.json /app/
 WORKDIR /app
+
+ADD package.json ./
+ADD package-lock.json ./
 RUN npm install
-ADD . /app
 
-CMD ["node", "bin/www"]
+ADD app app/
```

Comment:
So app from outside will end up in `/app/app`, is that what we want? Is there some convention that makes `npm start` start the app in `app`?

Reply:
Hang on, now I remember, it looks at the `package.json`...currently has:

```json
  "scripts": {
    "start": "node app/server.js"
  },
```

Is this relative to workdir?

Reply:
No it doesn't

Reply:
Hm, I read some best practice guidance somewhere that suggested `app/server.js` was the way to do it.
Having all the code in `app` makes it cleaner to add in the Dockerfile, though.

Reply:
Yes, you're right - in the docker container the app code is in `{WORKDIR}/app` and `WORKDIR` is `/app`. This isn't a problem - `WORKDIR` should probably be `/var/www` or `/srv` or something anyway.

Reply:
I think having the app in `/app` actually makes sense. It's nice that it's in the root directory and it's easy to spot (in case someone is not familiar with the image).